website/docs/storage.mdx

```diff
@@ -0,0 +1,199 @@
+
+### Block design
+
+In automerge, the list of text contains the blocks directly:
+
+```json
+[
+  { "id": "a1", "type": "paragraph" },
+  "h",
+  "e",
+  "l",
+  "l",
+  "o",
+  { "id": "a2", "type": "paragraph" }
+]
+```
+
+If there is nesting of blocks, like in lists, it will specify a `parent`:
+
+```json
+[
+  { "id": "a1", "type": "paragraph" },
+  "h",
+  "i",
+  { "id": "a2", "type": "emoji", "parent": "a1" }
+]
+```
+
+Our flat-pack format will store blocks sequentially in a list and use `\uFFFC` as placeholders. This means the first example will look like:
+
+```json
+{
+  "text": "\uFFFChello\uFFFC",
+  "blocks": [
+    { "id": "a1", "type": "paragraph" },
+    { "id": "a2", "type": "paragraph" }
+  ]
+}
+```
+
+When storing blocks in a database, the order that they occur in the document _must_ be stored in some manner, because the format specifically doesn't contain range information and instead uses a chunking algorithm to create boundaries from the object replacement characters (`\uFFFC`).
+
+### Text blocks
+
+In certain circumstances, there may be blocks that contain child blocks and continue after the end of the child block. There's other examples that create this "jagged" structure of block boundaries, and to solve this, atjson will introduce new block boundaries with a type of `"text"`. So the following document will produce a text block:
+
+```js
+let doc = new OffsetSource({
+  content: "Missy Elliott’s\n“Supa Dupa Fly”",
+  annotations: [
+    new LineBreak({ start: 15, end: 16 }),
+    new ParseAnnotation({ start: 15, end: 16 }),
+  ],
+});
+```
+
+import { serialize, ParseAnnotation } from "@atjson/document";
+import OffsetSource, {
+  LineBreak,
+  Paragraph,
+  Italic,
+  Bold,
+  List,
+  ListItem,
+} from "@atjson/offset-annotations";
+
+export function Serialized(props) {
+  return (
+    <code language="json">
+      {JSON.stringify(serialize(props.doc, props), null, 2)}
+    </code>
+  );
+}
+
+<Serialized
+  doc={
+    new OffsetSource({
+      content: "Missy Elliott’s\n“Supa Dupa Fly”",
+      annotations: [
+        new LineBreak({ start: 15, end: 16 }),
+        new ParseAnnotation({ start: 15, end: 16 }),
+      ],
+    })
+  }
+/>
+
+### Marks
+
+Marks are represented as an object with range information and attributes. Ranges contain information about edge behaviour that was previously handled by options on `insertText`.
+
+See [Peritext](https://www.inkandswitch.com/peritext/#generating-inline-formatting-operations) for context on boundary information with marks. If you wanted to line up this syntax with what Peritext uses, `[|]` is the same as `"before"` and `(|)` is the same as `"after"`. The brackets are commonly used in domain / ranges in math equations, and are reused here to coincide with how automerge stores this information.
```

Comment:
I think it would be good to motivate why we want to include the edge behavior in the storage format, which is not immediately obvious since that behavior is statically known based on the mark. This is necessary since we want to be able to merge documents based on the storage format alone - are there other reasons?